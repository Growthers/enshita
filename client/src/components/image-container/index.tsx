import type { FC } from "react";
import styles from "~/styles/components/image-container/image-container.module.scss";
import type { ImageContainerProperties } from "./type/model";

const ImageContainer: FC<ImageContainerProperties> = ({
  alt,
  children,
  src,
}) => (
  <div className={styles["enshita-image-container"]}>
    <div className={styles["enshita-image-container-image-wrapper"]}>
      <img
        src={src}
        alt={alt}
        className={styles["enshita-image-container-image"]}
      />
    </div>
    <p className={styles["enshita-image-container-content"]}>{children}</p>
  </div>
);

export default ImageContainer;
