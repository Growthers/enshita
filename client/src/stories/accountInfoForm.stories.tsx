// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStoryObj } from "@storybook/react";
import Form from "~/components/form/accountInfoForm";

type T = typeof Form;
type Story = ComponentStoryObj<T>;

export default {
  title: "Account Information Form",
  args: {
    mail: "abc@example.com",
    userName: "hoge",
  },
  component: Form,
} as ComponentMeta<T>;

export const Default: Story = {};
