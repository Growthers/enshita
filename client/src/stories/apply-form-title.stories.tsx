import type { ComponentStoryObj, ComponentMeta } from "@storybook/react";
import { Title } from "~/components/apply-form/title";

type T = typeof Title;
type Story = ComponentStoryObj<T>;

const data = {
  title: "共同開発鯖LT #4",
};

export default { args: { title: data.title }, component: Title } as ComponentMeta<T>;

export const Default: Story = {};
