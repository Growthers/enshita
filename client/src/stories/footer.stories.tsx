// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStoryObj } from "@storybook/react";
import Footer from "~/components/layout/footer";

type T = typeof Footer;
type Story = ComponentStoryObj<T>;

export default {
  title: "Footer",

  component: Footer,
} as ComponentMeta<T>;

export const Default: Story = {};
