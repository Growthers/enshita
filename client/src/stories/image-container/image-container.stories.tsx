import type { ComponentMeta, ComponentStoryObj } from "@storybook/react";
import ImageContainer from "~/components/image-container";

type T = typeof ImageContainer;
type Story = ComponentStoryObj<T>;

export default {
  args: {
    alt: "post new LT",
    children: "簡単にLTを\n開催できる.",
    src: "/statics/undraw_post_re_mtr4.svg"
  },
  component: ImageContainer,
} as ComponentMeta<T>;

export const Default: Story = {};
