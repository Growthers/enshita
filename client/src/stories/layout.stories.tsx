import React from "react";
// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStory } from "@storybook/react";
import Layout from "~/components/layout";

type T = typeof Layout;
type Story = ComponentStory<T>;

export default {
  title: "Layout",

  component: Layout,
} as ComponentMeta<T>;

export const Default: Story = () => (
  <Layout>
    <div>hogehoge</div>
  </Layout>
);
