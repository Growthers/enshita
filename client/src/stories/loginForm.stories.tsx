// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStoryObj } from "@storybook/react";
import LoginForm from "~/components/loginForm";

type T = typeof LoginForm;
type Story = ComponentStoryObj<T>;

export default {
  title: "LoginForm",
  component: LoginForm,
} as ComponentMeta<T>;

export const Default: Story = {};
