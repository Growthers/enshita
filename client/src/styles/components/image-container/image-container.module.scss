@use "../../variables" as var;

.enshita-image-container {
  background-color: var.$global-light-gray;
  border-radius: 0.5rem;
  box-shadow: 0 3px 0 0 var.$global-primary-pale;
  font-size: 1.25rem;
  color: var.$global-black;
  width: 14rem;
  padding: 3rem 1.5rem 2rem;
  & > :not([hidden]) ~ :not([hidden]) {
    margin-top: 2rem;
  }
  .enshita-image-container-image-wrapper {
    display: flex;
    justify-content: center;
    flex-grow: 0;
    height: 10rem;
    margin-bottom: 2rem;
    .enshita-image-container-image {
      width: 100%;
      display: flex;
    }
  }
  .enshita-image-container-content {
    white-space: pre-wrap;
    text-align: center;
  }
}
