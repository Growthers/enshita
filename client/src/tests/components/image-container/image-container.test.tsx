import { composeStories } from "@storybook/testing-react";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import * as stories from "~/stories/image-container/image-container.stories";

const { Default } = composeStories(stories);

const option = { name: "post new LT" };

describe("(components) image container for landing page", () => {
  test("to be [role=img]", () => {
    const { getByRole } = render(<Default />);
    expect(getByRole("img", option)).toBeInTheDocument();
  });
  test("take snapshot", () => {
    const { container } = render(<Default />);
    expect(container).toMatchSnapshot();
  });
});
