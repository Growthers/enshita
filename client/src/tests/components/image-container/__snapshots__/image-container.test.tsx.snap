// Vitest Snapshot v1

exports[`(components) image container for landing page > take snapshot 1`] = `
<div>
  <div
    class="_enshita-image-container_3533c2"
  >
    <div
      class="_enshita-image-container-image-wrapper_3533c2"
    >
      <img
        alt="post new LT"
        class="_enshita-image-container-image_3533c2"
        src="/statics/undraw_post_re_mtr4.svg"
      />
    </div>
    <p
      class="_enshita-image-container-content_3533c2"
    >
      簡単にLTを
開催できる.
    </p>
  </div>
</div>
`;
